src/engraving/rendering/dev/chordlayout.cpp

```diff
@@ -2722,8 +2722,20 @@ void ChordLayout::layoutChords3(const MStyle& style, const std::vector<Chord*>&
 void ChordLayout::getNoteListForDots(Chord* c, std::vector<Note*>& topDownNotes, std::vector<Note*>& bottomUpNotes,
                                      std::vector<int>& anchoredDots)
 {
-    bool hasVoices = c->measure()->hasVoices(c->staffIdx(), c->tick(), c->ticks());
-    if (!hasVoices) {
+    bool hasVoices = c->measure()->hasVoices(c->vStaffIdx(), c->tick(), c->ticks());
+    bool hasCrossNotes = c->staffMove();
+    Measure* m = c->measure();
+    if (!hasCrossNotes) {
+        for (size_t i = c->vStaffIdx() + 1 * VOICES; i < c->vStaffIdx() + 2 * VOICES; ++i) {
```

Comment:
It looks like some parentheses are missing here; adding `1 * VOICES` to something that is a staff index seems not a logical thing to do.
```suggestion
        for (size_t i = (c->vStaffIdx() + 1) * VOICES; i < (c->vStaffIdx() + 2) * VOICES; ++i) {
```
But: why would you look at the staff below the "visual staff" of the item in question? Note that cross-staff notation can be created not only by moving notes to the staff below, but also by moving to the staff above. And both can happen simultaneously in the case of a three-staff instrument: 
<img width="246" alt="Scherm­afbeelding 2023-10-12 om 18 29 35" src="https://github.com/musescore/MuseScore/assets/48658420/e03baff3-ec2e-4943-b59d-86b0f29e6ebf">

Reply:
The chord passed into `getNoteListForDots` is the top chord (lowest track number) on the stave, even if that top chord has been moved down from the stave above, so we can tell if that chord is cross staff easily.  This loop is to check through all the voices on the staff below the one we are laying out to see if any voices have been moved up.

Reply:
Although, when doing some performance testing I realised this wasn't confined to the current part.  It is now.